pkg/driver/config_dir_test.go

```diff
@@ -0,0 +1,304 @@
+	}
+}
+
+// TestInitConfigDirPreferredError asserts that a symlink an error is returned when a symlink cannot be created to the
+// preferredDir
+func TestInitConfigDirPreferredError(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create an empty legacy dir
+	legacyDir := filepath.Join(dir, "legacy")
+	if err := os.MkdirAll(legacyDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	// create the preferred dir
+	preferredDir := filepath.Join(dir, "preferred")
+	if err := os.MkdirAll(preferredDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	etcAmazonEfs := filepath.Join(dir, "bad", "path")
+
+	// function under test
+	if err := InitConfigDir(legacyDir, preferredDir, etcAmazonEfs); err == nil {
+		t.Errorf("Expected an error when calling InitConfigDir")
+	}
+}
+
+// TestInitConfigDirPreExistingConfigError asserts that an error is returned if a symlink cannot be created to the
+// legacyDir.
+func TestInitConfigDirPreExistingConfigError(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create legacy dir and a fake pre-existing conf file
+	legacyDir := filepath.Join(dir, "legacy")
+	if err := os.MkdirAll(legacyDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+	confFile := filepath.Join(legacyDir, "efs-utils.conf")
+	if f, err := os.Create(confFile); err != nil {
+		t.Fatalf("Unable to create file: %v", err)
+	} else {
+		f.Close()
+	}
+
+	// create the preferred dir which will go unused
+	preferredDir := filepath.Join(dir, "preferred")
+	if err := os.MkdirAll(preferredDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	etcAmazonEfs := filepath.Join(dir, "bad", "path")
+
+	// function under test
+	if err := InitConfigDir(legacyDir, preferredDir, etcAmazonEfs); err == nil {
+		t.Errorf("Expected an error when calling InitConfigDir")
+	}
+}
+
+// TestInitConfigDirDisappeared asserts that an error is returned if the preferred config dir is missing.
+func TestInitConfigDirDisappeared(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create an empty legacy dir
+	legacyDir := filepath.Join(dir, "legacy")
+	if err := os.MkdirAll(legacyDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	// do NOT create the preferred dir
+	preferredDir := filepath.Join(dir, "preferred")
+	etcAmazonEfs := filepath.Join(dir, "symlink")
+
+	// function under test
+	if err := InitConfigDir(legacyDir, preferredDir, etcAmazonEfs); err == nil {
+		t.Errorf("Expected an error when calling InitConfigDir")
+	}
+}
+
+// TestInitConfigDirDisappeared asserts that an error is returned if the preferred config dir is a file.
+func TestInitConfigDirIsFile(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create an empty legacy dir
+	legacyDir := filepath.Join(dir, "legacy")
```

Comment:
since most test cases need to create both a legacyDir and preferredDir (mimicking that kubernetes/kubelet would have created them before the Pod started since they are DirectoryOrCreate HostPaths), could be useful; to refactor it into a function